minigunapp/tasks.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from celery import shared_task
+from minigunapp.models import Email
+from django.core.mail import send_mail
+
+
+@shared_task
+def send_email(id):
+    """ Task that attempts to send the email with the given id.
+    Retries on a delay until max failures reached.
+    """
+
+    email = Email.objects.get(id=id)
+
+    try:
+        send_mail(
+            email.subject,
+            email.body,
+            email.from_,
+            [email.to],
+            fail_silently=False,
+        )
+        email.status = 'sent'
+        logging.info("Sent email {0}".format(email.id))
+    except Exception as ex:
+        logging.error("Failed to send email {0} ({1}): {2}"
+                      .format(email.id, email.retries. str(ex)))
+        email.status = 'error'
+        email.error_message = str(ex)
+        if(email.retries < 3):
+            logging.info("Queueing email {0} for retry".format(email.id))
+            email.retries += 1
+            email.save()
+            send_email.apply_async([str(email.id)], countdown=600)
+        return False
+
+    email.save()
```

Comment:
So if you're sending 100GB a day, a tenth of your disk space, you'd be in trouble before ten days had elapsed (some overhead from indexing and such). Some space management options:

* Build a database cluster and expand it aggressively, possibly with automatic horizontal scaling
* Siphon off "old" data to a larger, slower, long-term storage media, so it is accessible to dev/support/whatever, but not used directly by the app.
* Disk compression. Notable performance tradeoff here. Sacrificing disk I/O for space.
* Partition your dataset. Run your app in two different locales with two different infrastructures and route requests appropriately. So datacenter 1 could handle ops for data A-M and datacenter 2 could handle ops for data N-Z. In this case you're solving your space problem by building multiple database clusters.
* Make the app save less data. Try to audit what you're doing and eliminate redundancies and things you don't really need.